monitoring/uss_qualifier/scenarios/astm/utm/dss/subscription_validation.py

```diff
@@ -0,0 +1,212 @@
+from datetime import datetime, timedelta
+from typing import Dict, Any, List, Optional
+
+import loguru
+import s2sphere
+from s2sphere import LatLngRect
+from uas_standards.astm.f3548.v21.api import Subscription
+from uas_standards.astm.f3548.v21.constants import Scope
+
+from monitoring.monitorlib.geo import Polygon, Volume3D
+from monitoring.monitorlib.geotemporal import Volume4D
+from monitoring.monitorlib.mutate.scd import MutatedSubscription
+from monitoring.prober.infrastructure import register_resource_type
+from monitoring.uss_qualifier.resources import VerticesResource
+from monitoring.uss_qualifier.resources.astm.f3548.v21.dss import DSSInstanceResource
+from monitoring.uss_qualifier.resources.interuss.id_generator import IDGeneratorResource
+from monitoring.uss_qualifier.resources.uss_area import (
+    USSAreaResource,
+    USSAreaSpecification,
+)
+from monitoring.uss_qualifier.scenarios.astm.utm.dss import test_step_fragments
+from monitoring.uss_qualifier.scenarios.scenario import (
+    TestScenario,
+    PendingCheck,
+)
+from monitoring.uss_qualifier.suites.suite import ExecutionContext
+
+_24H_MIN_TOLERANCE_S = 23 * 3600 + 59 * 60  # 23 hours and 59 minutes
+_24H_MAX_TOLERANCE_S = 24 * 3600 + 1  # 24 hours sharp, plus a second
+
+
+class SubscriptionValidation(TestScenario):
+    """
+    A scenario that verifies that the DSS properly validates and sets the proper limits on subscriptions.
+    """
+
+    SUB_TYPE = register_resource_type(378, "Subscription")
+
+    # Base identifier for the subscriptions that will be created
+    _sub_id: str
+
+    _uss_area: USSAreaSpecification
+
+    _uss_area_volume4d: Volume4D
+
+    # Base parameters used for subscription creation variations
+    _sub_generation_params: Dict[str, Any]
+
+    def __init__(
+        self,
+        dss: DSSInstanceResource,
+        id_generator: IDGeneratorResource,
+        uss_area: USSAreaResource,
+    ):
+        """
+        Args:
+            dss: dss to test
+            id_generator: will let us generate specific identifiers
+            uss_area: An Area to use for the tests. It should be an area for which the DSS is responsible,
+                 but has no other requirements.
+        """
+        super().__init__()
+        scopes = {Scope.StrategicCoordination: "create and delete subscriptions"}
+        # This is an UTMClientSession
+        self._dss = dss.get_instance(scopes)
+        self._pid = [self._dss.participant_id]
+        self._sub_id = id_generator.id_factory.make_id(self.SUB_TYPE)
+        self._uss_area = uss_area.specification
+
+        # Keep the footprint, no altitudes
+        self._uss_area_volume4d = Volume4D(
+            volume=Volume3D(outline_polygon=Polygon(vertices=self._uss_area.footprint))
+        )
+
+        self._sub_generation_params = dict(
```

Comment:
nit: generally we want clear typing and so we [avoid plain dicts](https://github.com/interuss/tsc/blob/main/python_style_guide.md#data-structures).  Keyword argument dicts are a good exception to this rule, but in that case let's make it clear that's what we're doing

```suggestion
        self._sub_generation_kwargs = dict(
```